pkg/tombstone/tombstone.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package tombstone
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/cespare/xxhash"
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+const (
+	// TombstoneDir is the name of directory to upload tombstones.
+	TombstoneDir = "tombstones"
+)
+
+// Tombstone represents a tombstone.
+type Tombstone struct {
+	Matcher      string `json:"matcher"`
+	MinTime      int64  `json:"minTime"`
+	MaxTime      int64  `json:"maxTime"`
+	CreationTime int64  `json:"creationTime"`
+}
+
+// NewTombstone returns a new instance of Tombstone.
+func NewTombstone(matcher string, minTime, maxTime int64) Tombstone {
+	return Tombstone{
+		Matcher:      matcher,
+		MinTime:      minTime,
+		MaxTime:      maxTime,
+		CreationTime: timestamp.FromTime(time.Now()),
```

Comment:
Similar to Alertmanager Silencers wonder if we can allow specifying human readable reason why like `reason` field
and `author` field where people can optionally specify author of deletion.. :thinking: Maybe too much (: 

cc @metalmatze

Reply:
Added the author and reason flags :+1: